x-pack/plugins/observability_solution/apm/server/utils/es_fields_mappings/es_fields_mappings_helpers.ts

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import type { Fields } from './types';
+
+export const normalizeValue = <T>(field: unknown[] | unknown): T => {
+  return (Array.isArray(field) && field.length === 1 ? field[0] : field) as T;
+};
+
+export const isOptionalFieldDefined = (fields: Fields, fieldToCheck: string): boolean =>
+  Object.keys(fields)
+    .filter((field) => field.includes(fieldToCheck))
```

Comment:
I think the assumption from how this is being used is that the string starts with `fieldToCheck`, so this should reflect that

Reply:
That's fair, I changed it to `startsWith`